rules-reviewed/eap7/eap6/resteasy.windup.xml

```diff
@@ -0,0 +1,743 @@
+                    <message><![CDATA[
+                    Replace class `org.jboss.resteasy.client.ClientRequest` with `org.jboss.resteasy.client.jaxrs.ResteasyClient`.
+                    
+                    The following is an example of how to send a link header with the RESTEasy client in EAP 6 with RESTEasy 2.3.x.
+                    
+                    ```java
+                    ClientRequest request = new ClientRequest(generateURL("/linkheader/str"));
+                    request.addLink("previous chapter", "previous", "http://example.com/TheBook/chapter2", null);
+                    ClientResponse response = request.post();
+                    LinkHeader header = response.getLinkHeader();
+                    ```
+                    
+                    The following is an example of how to accomplish the same task with the RESTEasy client in RESTEasy 3.
+                    
+                    ```java
+                    ResteasyClient client = new ResteasyClientBuilder().build();
+                    Response response = client.target(generateURL("/linkheader/str")).request()
+                        .header("Link", "<http://example.com/TheBook/chapter2>; rel="previous";
+                    title="previous chapter">").post(Entity.text(new String()));
+                    javax.ws.rs.core.Link link = response.getLink("previous");
+                    ```
+                    
+                    ]]></message>
+                    <link title="RESTEasy Client Application Changes"
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/chapter-4-application-changes" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000003">
+            <when>
+                <javaclass references="org.jboss.resteasy.client.ClientResponse" />
+            </when>
+            <perform>
+                <hint title="Deprecated class ClientResponse in RESTEasy 3" effort="1" severity="optional">
+                    <message>Replace `org.jboss.resteasy.client.ClientResponse` class with `javax.ws.rs.core.Response`.</message>
+                    <link title="RESTEasy Client Application Changes"
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/chapter-4-application-changes" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000004">
+            <when>
+                <javaclass references="org.jboss.resteasy.client.{type}" />
+            </when>
+            <perform>
+                <hint title="Deprecated Client API class org.jboss.resteasy.client.{type} in RESTEasy 3" effort="1" severity="optional">
```

Comment:
No other rule for Client API classes use "Deprecated Client API class", only "Deprecated class".